### ISSUE TITLE
Add `typing-extensions` to dependencies

### ISSUE THREAD

**Issue by jlondonobo** [ToucanToco/fastexcel#226]
## What's the issue?
`fastexcel` uses `typing_extensions`:
https://github.com/ToucanToco/fastexcel/blob/81690535b05b14b5953c21c9e7fb77dfd915dcda/python/fastexcel/__init__.py#L5

However, `typing_extensions` is not part of dependencies in [`pyproject.toml` ](https://github.com/ToucanToco/fastexcel/blob/81690535b05b14b5953c21c9e7fb77dfd915dcda/pyproject.toml)

This causes a `ModuleNotFoundError` when `fastexcel` is installed by itself.

## How to reproduce the bug?

1. Create a virtual environment:
```bash
python -m venv .venv/
source .venv/bin/activate
```

2. Install `fastexcel`:
```bash
pip install -e .
python
```

3. Import `fastexcel`
```python
import fastexcel
```

## Error message
```bash
Python 3.12.2 (main, Feb 20 2024, 04:06:49) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import fastexcel
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/joselondono/Documents/forks/fastexcel/python/fastexcel/__init__.py", line 5, in <module>
    from typing_extensions import TypeAlias
ModuleNotFoundError: No module named 'typing_extensions'
```

**Comment by PrettyWood** [comment]
Hello @jlondonobo 
Thank you for the great report! You're absolutely right.

**Comment by PrettyWood** [comment]
closed by #227
